package.json

```diff
@@ -150,129 +150,126 @@
-        },
-        "terraform-ls.ignoreDirectoryNames": {
-          "scope": "resource",
-          "type": "array",
-          "items": {
-            "type": "string"
+          "terraform.languageServer.trace.server": {
+            "scope": "window",
+            "type": "string",
+            "enum": [
+              "off",
+              "messages",
+              "verbose"
+            ],
+            "default": "off",
+            "description": "Traces the communication between VS Code and the language server."
           },
-          "default": [],
-          "description": "Per-workspace list of directory names for the language server to ignore when indexing files"
-        },
-        "terraform-ls.experimentalFeatures": {
-          "scope": "resource",
-          "type": "object",
-          "description": "Experimental (opt-in) terraform-ls features",
-          "properties": {
-            "validateOnSave": {
-              "scope": "resource",
-              "type": "boolean",
-              "default": false
+          "terraform-ls.rootModules": {
+            "scope": "resource",
+            "type": "array",
+            "items": {
+              "type": "string"
             },
-            "prefillRequiredFields": {
-              "scope": "resource",
-              "type": "boolean",
-              "default": false
+            "default": [],
+            "description": "Per-workspace list of module directories for the language server to read"
+          },
+          "terraform-ls.terraformExecPath": {
+            "scope": "resource",
+            "type": "string",
+            "description": "Path to the Terraform binary"
+          },
+          "terraform-ls.terraformExecTimeout": {
+            "scope": "resource",
+            "type": "string",
+            "description": "Overrides Terraform execution timeout (e.g. 30s)"
+          },
+          "terraform-ls.terraformLogFilePath": {
+            "scope": "resource",
+            "type": "string",
+            "description": "Path to a file for Terraform executions to be logged into (TF_LOG_PATH) with support for variables (e.g. timestamp, pid, ppid) via Go template syntax {{varName}}"
+          },
+          "terraform-ls.excludeRootModules": {
+            "scope": "resource",
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "default": [],
+            "description": "Per-workspace list of module directories for the language server to exclude"
+          },
+          "terraform-ls.ignoreDirectoryNames": {
+            "scope": "resource",
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "default": [],
+            "description": "Per-workspace list of directory names for the language server to ignore when indexing files"
+          },
+          "terraform-ls.experimentalFeatures": {
```

Comment:
Am I right that there isn't much we can do about making this one editable (without breaking stuff)?
![Screenshot 2022-04-05 at 16 02 32](https://user-images.githubusercontent.com/287584/161784379-59b585c8-142b-47e8-83ea-cd859df5afe2.png)

Reply:
I have some ideas. Let me test and get back to you.